astropy/units/core.py

```diff
@@ -939,7 +942,6 @@ def __getstate__(self):
         # hashes of strings vary between sessions.
         state = self.__dict__.copy()
         state.pop("_hash", None)
-        state.pop("_type_id", None)
```

Comment:
We should not remove this but instead change it to `_physical_type_id`. Recall that `cached_property` stores the actual result in `__dict__` using the same name - this state should not be kept (because, as the comment notes, hashes of strings vary between sessions).

Reply:
Is there any hope of having a test for that?

Reply:
I guess one would have to write a test that does multiple, independent python runs at the same time, and communicates between them, not sure...